apps/token/src/routes/governance/components/propose/proposal-form-vote-and-enactment-deadline.tsx

```diff
@@ -0,0 +1,380 @@
+import { useEffect, useState, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { parse as ISO8601Parse, toSeconds } from 'iso8601-duration';
+import {
+  ButtonLink,
+  FormGroup,
+  Input,
+  InputError,
+} from '@vegaprotocol/ui-toolkit';
+import { addHours, addMinutes } from 'date-fns';
+import { ProposalFormSubheader } from './proposal-form-subheader';
+import type { UseFormRegisterReturn } from 'react-hook-form';
+
+export interface ProposalFormVoteAndEnactmentDeadlineProps {
+  voteRegister: UseFormRegisterReturn<'proposalVoteDeadline'>;
+  voteErrorMessage: string | undefined;
+  voteMinClose: string;
+  voteMaxClose: string;
+  enactmentRequired?: boolean;
+  enactmentRegister?: UseFormRegisterReturn<'proposalEnactmentDeadline'>;
+  enactmentErrorMessage?: string;
+  enactmentMinClose?: string;
+  enactmentMaxClose?: string;
+  validationRequired?: boolean;
+  validationRegister?: UseFormRegisterReturn<'proposalValidationDeadline'>;
+  validationErrorMessage?: string;
+}
+
+export const ProposalFormVoteAndEnactmentDeadline = ({
+  voteRegister,
+  voteErrorMessage,
+  voteMinClose,
+  voteMaxClose,
+  enactmentRegister,
+  enactmentErrorMessage,
+  enactmentMinClose,
+  enactmentMaxClose,
+  validationRequired,
+  validationRegister,
+  validationErrorMessage,
+}: ProposalFormVoteAndEnactmentDeadlineProps) => {
+  const {
```

Comment:
chore (non-blocking): this is pretty hard to read, could we perhaps move the logic into it's own hook to separate out the UI and logic parts of the component